### PR TITLE
Code cleanup

### PR DESCRIPTION
**Pull request by Uberi**
Make all the dashboard code conform better to the [style guide](https://developer.mozilla.org/en-US/docs/Mozilla/Developer_guide/Coding_Style#JavaScript_practices).

There are also a few semantic changes to make the code a bit neater, but the functionality should remain exactly the same.

### CODE REVIEW

File: new-pipeline/src/dashboards.js [hunk 1]
```diff
@@ -149,67 +167,116 @@ function loadStateFromUrlAndCookie() {
...
 
   // Obtain a mapping from filter names to filter options
   var filterMapping = {};
   for (var filterName in filters) {
+    // Some options are not selected, so we need to explicitly filter
     var selector = filters[filterName];
     var selected = selector.val() || [];
-    if (selected.length !== selector.find("option").length) { // Some options are not selected, so we need to explicitly filter
+    if (selected.length !== selector.find("option").length) {
       if (filterName === "os") { selected = compressOSs(); }
```
Comment (vdjeric) [inline new-pipeline/src/dashboards.js]: don't do one-line if's, put the statement on a line below the if

File: new-pipeline/src/dashboards.js [hunk 2]
```diff
@@ -404,19 +485,24 @@ function getHumanReadableOptions(filterName, options) {
...
     // Move the latest version of each channel into their own section
     var latest = {};
     goodOptions.forEach(function(option) {
       var parts = option.split("/");
-      if (!latest.hasOwnProperty(parts[0]) || (parseInt(parts[1]) > latest[parts[0]] && parseInt(parts[1]) < 70)) {
-        latest[parts[0]] = parseInt(parts[1]);
-      }
+      var optionIsLater = !latest.hasOwnProperty(parts[0]) ||
+        (parseInt(parts[1]) > latest[parts[0]] && parseInt(parts[1]) < 70);
+      if (optionIsLater) { latest[parts[0]] = parseInt(parts[1]); }
```
Comment (vdjeric) [inline new-pipeline/src/dashboards.js]: one-line if

File: new-pipeline/src/dist.js [hunk 3]
```diff
@@ -206,105 +318,145 @@ function calculateHistograms(callback, sanitize) {
...
+
+            // Make a list of evolutions and option labels for each label in the evolution
+            for (var label in fullEvolutionMap) {
+              if (sanitize) { fullEvolutionMap[label] = fullEvolutionMap[label].sanitized(); }
+              if (fullEvolutionMap[label] !== null) {
+                if (!fullEvolutionsMap.hasOwnProperty(label)) { fullEvolutionsMap[label] = []; }
+                fullEvolutionsMap[label].push(fullEvolutionMap[label]);
               }
-            }, dates, false);
+              if (!optionValues.hasOwnProperty(label)) { optionValues[label] = []; }
```
Comment (vdjeric) [inline new-pipeline/src/dist.js]: one-line if's.. just grep for them in the next pull

File: v2/telemetry.js [hunk 4]
```diff
@@ -268,114 +330,165 @@ Telemetry.init = function Telemetry_init(callback) {
...
+        if (Telemetry.CHANNEL_VERSION_BUILDIDS[channel][version]) {
+          versions.push(channel + "/" + version);
+        }
       }
     }
   }
+
   versions.sort();
-  return fromVersion !== undefined ? versions.filter(function(version) { return fromVersion <= version && version <= toVersion }) : versions;
+  return fromVersion !== undefined ? versions.filter(function(version) {
```
Comment (vdjeric) [inline v2/telemetry.js]: this return is way too fancy. break it up with some if's and temporary variables

File: new-pipeline/src/dashboards.js [hunk 5]
```diff
@@ -149,67 +167,116 @@ function loadStateFromUrlAndCookie() {
...
+    pageState.min_channel_version.indexOf("/") >= 0
+  ) ? pageState.min_channel_version : null;
+  pageState.max_channel_version = (
+    typeof pageState.max_channel_version === "string" &&
+    pageState.max_channel_version.indexOf("/") >= 0
+  ) ?  pageState.max_channel_version : null;
+  pageState.product = (
+    typeof pageState.product === "string" &&
+    pageState.product !== "" && pageState.product !== "null"
+  ) ? pageState.product.split("!").filter(function(v) { return v !== ""; }) : ["Firefox"];
```
Comment (vdjeric) [inline new-pipeline/src/dashboards.js]: the over-use of the conditional operator is making the code super-dense and hard to read, replace with ifs whenever it's a statement more complex than "x = (a + 2 > b) ? 1 : 0"

**Comment by vdjeric** [comment]
@Uberi Do I still need to merge this?

**Comment by Uberi** [comment]
There's a lot of conflicts... might be better to redo with latest changes.

**Comment by vdjeric** [comment]
Let me know when the code clean up pull is ready

**Comment by Uberi** [comment]
Reformatted using JSBeautifier, plus several manual formatting tweaks to make things look a bit better.

**Pull request merged**